### ISSUE TITLE
Search: post count should use the same API

### ISSUE THREAD

**Issue by kangzj** [Automattic/jetpack#24684]
### Impacted plugin

Jetpack, Search

### What

post count should use the same API between sites and checkout

### How

- Create an API to expose record estimation
- Call the record estimation from WPCOM checkout
 
 For now it's a similar but different process, which could cause discrepancies.

**Comment by trakos** [comment]
I believe that between https://github.com/Automattic/jetpack/pull/40635 and 169911-ghe-Automattic/wpcom we have now resolved this issue as well.

cc @gibrown and @oikeme in case you think there's more to do and we shouldn't close this just yet.

**Comment by gibrown** [comment]
Yep. I think this is fixed. We can see if any of these post count issues come up again in the future, but we've improved a lot.
